app/lib/features/deep_linking/actions/handle_deep_link_uri.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:acter/features/deep_linking/types.dart';
+import 'package:acter/features/deep_linking/utils.dart';
+import 'package:acter/features/super_invites/dialogs/redeem_dialog.dart';
+import 'package:acter/features/users/actions/show_global_user_dialog.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+Future<void> handleDeepLinkUri({
+  required BuildContext context,
+  required WidgetRef ref,
+  required Uri uri,
+}) async {
+  final lang = L10n.of(context);
+  try {
+    final result = parseUri(uri);
+    final _ = switch (result.type) {
```

Comment:
otherwise my IDE complaints about the new `switch`-statement... am pretty sure this is just a false-positive lint flagging because this is a pretty new feature..